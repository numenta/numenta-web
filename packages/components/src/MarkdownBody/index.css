/**
 * Numenta Web Platform and Sites source code
 * MIT License (see LICENSE.txt)
 * Copyright © 2005—2017 Numenta <http://numenta.com>
 */

._dropcap > div > p:first-of-type {
  margin-top: 0;
  padding-top: 0;
}

/* dropcap - sync with wrappers/md/md.css */
._dropcap > div > p:first-of-type::first-letter {
  background-color: #fff;
  color: #1aa0db;
  float: left;
  font-weight: 500;
  line-height: 0.85;
  margin: 0 0 -1rem 0;
  /* -s (<= mobile) */
  font-size: 6.8rem;
  padding: 0.5rem 0.3rem 0.3rem 0;
}
@media screen and (min-width: 48em) {
  ._dropcap > div > p:first-of-type::first-letter {
    /* -ns (> mobile) */
    font-size: 5.4rem;
    padding: 0.4rem 0.3rem 0.3rem 0;
  }
}

.markdownBody {
  composes: _dropcap;
}
