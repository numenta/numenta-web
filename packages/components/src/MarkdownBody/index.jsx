// Numenta Web Platform and Sites source code
// MIT License (see LICENSE.txt)
// Copyright © 2005—2017 Numenta <http://numenta.com>

import React from 'react'

import Markdown from '../Markdown'

import styles from './index.css'


/**
 * Markdown Body item text wrapper for lone Markdown text columns.
 */
const MarkdownBody = ({markdown}) => {
  const {body} = markdown

  return (
    <Markdown>
      <div className={styles.markdownBody}>
        <div dangerouslySetInnerHTML={{__html: body}} />
      </div>
    </Markdown>
  )
}

MarkdownBody.propTypes = {
  markdown: React.PropTypes.object.isRequired,
}

export default MarkdownBody
