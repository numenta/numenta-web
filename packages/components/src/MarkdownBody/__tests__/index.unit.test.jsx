import React from 'react'
import renderer from 'react-test-renderer'

import MarkdownBody from '../../MarkdownBody'

const markdown = {
  body: '# Hello world  ## Whadup',
}


describe('MarkdownBody React component', () => {

  it('Renders correctly', () => {
    const component = renderer.create(
      <MarkdownBody markdown={markdown} />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

})
