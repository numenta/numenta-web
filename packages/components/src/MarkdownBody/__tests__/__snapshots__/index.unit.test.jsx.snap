exports[`MarkdownBody React component Renders correctly 1`] = `
<div
  className="markdown">
  <div
    className="markdownBody">
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": "# Hello world  ## Whadup",
        }
      } />
  </div>
</div>
`;
